CRM/Activity/BAO/Activity.php

```diff
@@ -2772,7 +2730,7 @@ public static function checkPermission($activityId, $action) {
     $targetID = CRM_Utils_Array::key('Activity Targets', $activityContacts);
 
     // Check for source contact.
-    if (!$componentId || $allow) {
```

Comment:
@eileenmcnaughton this seems to be a bit odd given that what if there is no component id?

Reply:
@seamuslee001 yeah I remove componentId there because $componentId is only ever true when allow is also true